be/src/exec/schema_scanner/schema_tablets_scanner.cpp

```diff
@@ -0,0 +1,220 @@
+        {"ROW_SIZE", TYPE_BIGINT, sizeof(int64_t), true},
+        {"COMPACTION_SCORE", TYPE_INT, sizeof(int32_t), true},
+        {"COMPRESS_KIND", TYPE_STRING, sizeof(StringRef), true},
+        {"IS_USED", TYPE_BOOLEAN, sizeof(bool), true},
+        {"IS_ALTER_FAILED", TYPE_BOOLEAN, sizeof(bool), true},
+        {"CREATE_TIME", TYPE_DATETIME, sizeof(int64_t), true},
+        {"UPDATE_TIME", TYPE_DATETIME, sizeof(int64_t), true},
+        {"IS_OVERLAP", TYPE_BOOLEAN, sizeof(bool), true},
+};
+
+SchemaTabletsScanner::SchemaTabletsScanner()
+        : SchemaScanner(_s_tbls_columns, TSchemaTableType::SCH_BACKEND_TABLETS) {};
+
+Status SchemaTabletsScanner::start(RuntimeState* state) {
+    if (!_is_init) {
+        return Status::InternalError("used before initialized.");
+    }
+    _backend_id = state->backend_id();
+    RETURN_IF_ERROR(_get_all_tablets());
+    return Status::OK();
+}
+
+Status SchemaTabletsScanner::_get_all_tablets() {
+    if (config::is_cloud_mode()) {
+        auto tablets =
+                ExecEnv::GetInstance()->storage_engine().to_cloud().tablet_mgr().get_all_tablet();
+        std::ranges::for_each(tablets, [&](auto& tablet) {
+            _tablets.push_back(std::static_pointer_cast<BaseTablet>(tablet));
+        });
+    } else {
+        auto tablets = ExecEnv::GetInstance()
+                               ->storage_engine()
+                               .to_local()
+                               .tablet_manager()
+                               ->get_all_tablet();
+        std::ranges::for_each(tablets, [&](auto& tablet) {
+            _tablets.push_back(std::static_pointer_cast<BaseTablet>(tablet));
+        });
+    }
+    return Status::OK();
+}
+
+Status SchemaTabletsScanner::get_next_block_internal(vectorized::Block* block, bool* eos) {
+    if (!_is_init) {
+        return Status::InternalError("Used before initialized.");
+    }
+    if (nullptr == block || nullptr == eos) {
+        return Status::InternalError("input pointer is nullptr.");
+    }
+    *eos = true;
+    return _fill_block_impl(block);
+}
+
+Status SchemaTabletsScanner::_fill_block_impl(vectorized::Block* block) {
+    SCOPED_TIMER(_fill_block_timer);
+
+    size_t row_num = _tablets.size();
+    if (row_num == 0) {
+        return Status::OK();
+    }
+
+    size_t fill_tablets_num = _tablets.size();
+    std::vector<void*> datas(fill_tablets_num);
+
+    auto fill_column = [&](auto&& get_value, size_t column_index) {
+        using ValueType = std::decay_t<decltype(get_value(std::declval<BaseTabletSPtr>()))>;
+        std::vector<ValueType> srcs(fill_tablets_num);
+        for (size_t i = 0; i < _tablets.size(); ++i) {
+            BaseTabletSPtr tablet = _tablets[i];
+            srcs[i] = get_value(tablet);
+            datas[i] = &srcs[i];
+        }
+        return fill_dest_column_for_range(block, column_index, datas);
+    };
+
+    auto fill_boolean_column = [&](auto&& get_value, size_t column_index) {
```

Comment:
I find another question, every time calling fill_boolean_column, then _tablets collection is traversed once.
If a BE has 500000 tablets, then the tablets colleciton could be traversed 500000 * schema.size() times, it costs too much.